pyfar/dsp/dsp_interpolation.py

```diff
@@ -0,0 +1,731 @@
+import numpy as np
+from scipy.special import iv as bessel_first_mod
+from scipy.interpolate import interp1d
+import scipy.signal as sgn
+import matplotlib.pyplot as plt
+import pyfar as pf
+
+
+def smooth_fractional_octave(signal, width, mode="magnitude",
+                             window="boxcar"):
+    """
+    Smooth spectrum with a fractional octave width.
+
+    The smoothing is done according to Tylka et al. 2017 [#]_ method 2 in three
+    steps:
+
+    1. Interpolate the spectrum to a logarithmically spaced frequency scale
+    2. Smooth the spectrum by convolution with a smoothing window
+    3. Interpolate the spectrum to the original linear frequency scale
+
+    Note: Method 3 in Tylka at al. 2017 is mathematically more elegant at the
+    price of a largely increased computational and memory cost. In most
+    practical cases, methods 2 and 3 yield close to identical results (cf. Fig.
+    2 and 3 in Tylka et al. 2017). If the spectrum contains extreme
+    discontinuities, however, method 3 is superior (see examples below).
+
+    Parameters
+    ----------
+    signal : pyfar.Signal
+        The input data.
+    width : number
+        The width of the smoothing window in octaves, e.g., 1/3 will apply
+        third octave smoothing.
+    mode : str, optional
+
+        ``"magnitude"``
+            Only the magnitude response, i.e., the absolute spectrum is
+            smoothed. Note that this return as zero-phase signal. It might be
+            necessary to generate a minimum or linear phase if the data is
+            subject to further processing after the smoothing (cf.
+            :py:func:`~pyfar.dsp.minimum_phase` and
+            :py:func:`~pyfar.dsp.linear_phase`)
+        ``"magnitude_phase"``
+            Separately smooth the magnitude and unwrapped phase response
+        ``"magnitude_copy"``
+            Smooth the magnitude and keep the phase of the input signal.
+        ``"complex"``
+            Separately smooth the real and imaginary part of the spectrum. This
+            method often causes artifacts at high frequencies.
+
+        The default is ``"magnitude"``.
+    window : str, optional
+        String that defines the smoothing window. All window from
+        :py:func:`~pyfar.dsp.time_window` that does not require an additional
+        parameter can be used. The default is "boxcar", which uses the
+        most commonly used rectangular window.
+
+    Returns
+    -------
+    signal : pyfar.Signal
+        The smoothed output data
+    window_stats : tuple
+        A tuple containing information about the smoothing process
+
+        `n_window`
+            The window length in (logarithmically spaced) samples
+        `window_width`
+            The actual window width in octaves. This can deviate due to the
+            ideal width because the window must have an integer length
```

Comment:
If `width` is adjusted to be a fraction, it would make sense to change `window_width` accordingly

Reply:
see above - I'd rather leave this as it is

Reply:
Yes, I did that!